src/GraphODataTemplateWriter/Extensions/OdcmModelExtensions.cs

```diff
@@ -292,6 +294,103 @@ public static OdcmProperty GetServiceCollectionNavigationPropertyForPropertyType
+            var navigatedTypes = new HashSet<string>();
+            var matchingRootProperty = GetOdcmNamespaces(model)
+                .SelectMany(
+                    @namespace =>
+                        @namespace
+                            .Classes
+                            .FirstOrDefault(odcmClass => odcmClass.Kind == OdcmClassKind.Service)           // find the root service class
+                            ?.Properties
+                            ?.Where(serviceProperty => serviceProperty.GetType() == typeof(OdcmSingleton))  // find the singleton properties
+                            ?.Where(singleton =>
+                                        singleton.
+                                                Type.AsOdcmClass()
+                                                ?.Properties
+                                                ?.Where(property => property.ContainsTarget) // find singleton type properties that are self contained 
+                                                ?.Any(property => property.IsPropertyTypeChainedContainedNavigationProperty( //  keep following the containsTarget=True to find if we can use a reference
+                                                                                odcmProperty,
+                                                                                navigatedTypes,
+                                                                                property.IsCollection
+                                                                                    ? $"{singleton.Name}/{property.Name}/{{id}}"
+                                                                                    : $"{singleton.Name}/{property.Name}")
+                                                ) ?? false
+                            ) ?? Array.Empty<OdcmProperty>()
+                    ).FirstOrDefault();
+
+            return matchingRootProperty != null;
+        }
+
+        /// <summary>
+        /// This method takes in a property and looks through to see if the given testProperty could be contained in it through its contained navigation properties or
+        /// its nested contained navigation properties.
+        /// </summary>
+        /// <param name="odcmRootProperty">The property to start the search from</param>
+        /// <param name="testProperty">The property to find in the from the root type</param>
+        /// <param name="navigatedTypes">The types already navigated to prevent circular references</param>
+        /// <param name="route">The route followed to get here</param>
+        /// <returns></returns>
+        private static bool IsPropertyTypeChainedContainedNavigationProperty(this OdcmProperty odcmRootProperty, OdcmProperty testProperty, HashSet<string> navigatedTypes, string route)
+        {
+            // check if the property already matches the type.
+            if (odcmRootProperty.Type.FullName.Equals(testProperty.Type.FullName))
```

Comment:
```suggestion
            if (odcmRootProperty.Type.FullName.Equals(testProperty.Type.FullName, StringComparison.OrdinalIgnoreCase))
```

Reply:
Thanks, this is now resolved via https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/596/commits/a0a850149876d7f613aede4c228677f706a63978